### ISSUE TITLE
change drill-down handling of output_of

### ISSUE THREAD

**Issue by mah11** [pombase/canto#1307]
Curators have come to the disquieting realization that Canto should change how it presents terms that are linked by the *output_of* relation. I think only FYPO is affected at present.

What it does now: offer the output if you search for the thing on the other side of the link. For example, FYPO has the link 'abnormal Golgi morphology' output_of 'abnormal Golgi organization'. For 'abnormal Golgi organization' Canto offers 'abnormal Golgi morphology', whereas if I start with 'abnormal Golgi morphology' it doesn't offer 'abnormal Golgi organization'.

What we want it to do: basically the other way round - offer the output if we start from the thing that may have the output. In the Golgi example, we want to start with 'abnormal Golgi morphology' and have it offer 'abnormal Golgi organization'.

This might require the term/link swap described in https://github.com/pombase/canto/issues/590#issuecomment-119428858 (part of a long, convoluted discussion mainly about has_part).

This has come up before, and at the time we closed #803, Val thought all was well. I don't think Canto's behavior has changed since then; instead, we've reached a better understanding of what we want it to do, which calls for a change. It may not be urgent, since we've lived with the present behavior for so long.

Happy to chat if this needs further or better explanation. It's a head-melter.

**Comment by kimrutherford** [comment]
I think I will need a Skype chat about this at some point.

**Comment by mah11** [comment]
I am exactly the opposite of surprised by that. Maybe at the next group chat we can decide how soon we want this done, and schedule the chat-for-this accordingly.

**Comment by ValWood** [comment]
Hmm, I was so sure that we checked and check this and it was right. I hope its not one of those weird things that different  classes of phenotype need a different interpretation. I'd like to check a few if possible. If there is an easy way to provide me the list (or even a partial list) of some related terms, I can see if they make sense or not in the current Canto.

and I'll put this on the list for the next discussion slot....

**Comment by ValWood** [comment]
I'm putting this as high priority because it could easily cause a lot of incorrect term selection at leaf nodes if it is the wrong way round....

**Comment by mah11** [comment]
OK, the attached file has all of the output_of links in FYPO as of now.
[output_of_2017-02-06.txt](https://github.com/pombase/canto/files/754134/output_of_2017-02-06.txt)

**Comment by ValWood** [comment]
we need to do this, but I need to  evaluate current behaviour more thoroughly.
Today hopefully, or maybe tomorrow.....

v

**Comment by ValWood** [comment]
Hmm, based on this, I think the current behaviour in canto is correct?

https://github.com/pombase/pombase-chado/issues/613

I checked a lot of examples from the list of output_of. I did not find any cases where I would not expect to see the term presented/ inference this way around.....

**Comment by ValWood** [comment]
@mah11  could you look at my examples?
Why did we think it was wrong, I can't remember that either!

**Comment by mah11** [comment]
Remember that this ticket is about terms to offer in Canto, not annotation inferences. We don't necessarily expect Canto drill-down to go in the same direction as inference - think of is_a or part_of, where there's a clear difference in specificity. Inferring annotations goes from more specific to less specific, i.e. if a gp is annotated to a more specific term (nucleolus or mitotic DNA replication initiation), annotation to the less specific term is inferred (nucleus, DNA replication). But in Canto the drill-down goes in the opposite direction, from less specific to more specific. You search for nucleus, and then decide whether to use nucleolus or nucleoplasm etc., or not use any of the more specific suggestions.

It took a while to think through output_of, because there often isn't such an obvious specificity difference between the different terms, but I eventually figured out that Canto should start with the output and offer the thing (usually an abnormal process) that it's an output of.

So using the example above, you see that the Golgi looks wonky, and reach 'abnormal Golgi morphology'. You should then be offered 'abnormal Golgi organization', and consider whether your experiment actually watched the Golgi organization process taking place (and going wrong), or if you only saw a static dodgy Golgi. That's not what Canto is doing right now, so it should change.

tl;dr: The initial summary here is correct.

**Comment by ValWood** [comment]
The drill down  in Canto is behaving the way I expected at the moment.
See the examples in this ticket:
https://github.com/pombase/pombase-chado/issues/613

I followed these through Canto, and query builder, and a number of other examples. All are currently behaving as I expected, for both. I think morphology terms should be presented as "suggestions" for organization.
You observe a morhology defect, and abnormal organization can be always be inferred from this observation, so it isn't necessary to see "abnormal organization" in drill down if you select "abnormal morphology",  (because abnormal organization will be inferred if you select "abnormal morphology").

Isn't that exactly the desired behaviour?

**Comment by ValWood** [comment]
Note that its probably clearer with other examples than the morphology-> organization connection examples. Have a look at some of the others, it always works for me....

**Comment by ValWood** [comment]
So for example in Canto

increased interkinetochore distance before mitotic anaphase	FYPO:0002639	output_of	abnormal centromere localization	FYPO:0001778


we see 
increased interkinetochore distance before mitotic anaphase
as a suggestion if we search on 
abnormal centromere localization	FYPO:0001778


We want people to be encouraged to curate the "observation"
increased interkinetochore distance before mitotic anaphase
but we would not want them to 
see

abnormal centromere localization	FYPO:0001778
as a suggestion for 
increased interkinetochore distance before mitotic anaphase

because it is more "general" than the actual phenotype observation...

**Comment by mah11** [comment]
I disagree, because I think we should choose FYPO terms in Canto based on what was actually observed directly. Some experiments can show abnormal morphology but do not explicitly assay the organization process.

If you want another example, a static observation can tell you a cell has increased interkinetochore distance before mitotic anaphase, but you have to take a movie to see abnormal centromere localization.

If you think abnormal centromere localization is less specific than the process that causes increased interkinetochore distance before mitotic anaphase, then maybe that particular link shouldn't be in the ontology.

That said, I don't use the drill-down at all, so my preference is kind of irrelevant. If you're happy with someone annotating to a process phenotype based on a static image, we can leave the behavior unchanged.

**Comment by ValWood** [comment]
This is my point....the purpose is to encourage people to annotate to the observation.

I went through every output of in the list and   a) the inference and b) the specificity of annotation from drill down, made sense to me the way it is now.

What I want to avoid, is   a user to see , for example, 

increased interkinetochore distance before mitotic anaphase
and from there see the suggestion:
abnormal centromere localization
(which is less specific, in that everything annotated to increased interkinetochore distance before mitotic anaphase can be used to infer abnormal centromere localization, centromere location is abnormal in these mutants)

People infer  "abnormal centromere localization" from "increased interkinetochore distance before mitotic anaphase" . If they search on "abnormal centromere localization" I want them to see, and use "increased interkinetochore distance before mitotic anaphase" instead, so that they provide more *accurate* annotation based on their observation.

The output_of links are used in Canto to prevent the user from annotating to the "process phenotype" and to choose the *actual* observation instead.

I think it is safe to have has_output links to infer  abnormal processes, if a particular observation is always linked to a specific abnormal process.

For me it has been working so far....

The morphology-> organization  connection might be odd if you cannot always 
assume abnormal organization from abnormal morphology. I assumed that you could infer this because in GO morphogeneis is a descendant of organization?

**Comment by mah11** [comment]
The basis for my preference is that there are more kinds of experiment that let you see "increased interkinetochore distance before mitotic anaphase" than will let you observe "abnormal centromere localization" as it happens.

> The output_of links are used in Canto to prevent the user from annotating to the "process phenotype" and to choose the actual observation instead.

To me this seems like it would work better if the process phenotype wasn't the one offered first. But I have already said that my preference need not prevail, especially since we have so little data on how community users really use Canto.

> I think it is safe to have has_output links to infer abnormal processes,

No. Inference seems to be OK over output_of, but with has_output inference goes from the process to the structure.

 A output_of B = when A exists or happens, it's as a result of B. It's safe to infer annotation to B from annotation to A.

 A has_output B = if A happens, B always results. But there may be another way to get B, so if you have an annotation to B it's not safe to infer annotation to A.

For some pairs it may be correct to say both A output_of B and B has_output A, although done so anywhere in FYPO.

> ... assume abnormal organization from abnormal morphology. I assumed that you could infer this because in GO morphogenesis is a descendant of organization?

Not really; it's that I've felt safe inferring abnormal organization from abnormal morphology because I haven't thought of another way to get abnormal morphology besides organization going wrong.

**Comment by ValWood** [comment]
> To me this seems like it would work better if the process phenotype wasn't the one offered first. But I have already said that my preference need not prevail, especially since we have so little data on how community users really use Canto.

I don't understand this. Its about what people search on. They are likely to search on
"abnormal centromere localization", which is what they infer. Then they will find
"increased interkinetochore distance before mitotic anaphase"
and use that instead.

This is really just about getting the user to select the most appropriate term.



> The basis for my preference is that there are more kinds of experiment that let you see "increased interkinetochore distance before mitotic anaphase" than will let you observe "abnormal centromere localization" as it happens.

I don't understand this comment.

**Comment by mah11** [comment]
> > The basis for my preference is that there are more kinds of experiment that let you see "increased interkinetochore distance before mitotic anaphase" than will let you observe "abnormal centromere localization" as it happens.
> 
> I don't understand this comment.

With these example phenotypes, you can see increased distance between kinetochores from a single still picture, or from a movie or kymograph. But the still picture isn't a direct observation of an abnormal process of centromere localization taking place. It's only a static observation of a consequence of the abnormal process.

**Comment by mah11** [comment]
> This is really just about getting the user to select the most appropriate term.

Which term is most appropriate depends on what experiment was done.

**Comment by ValWood** [comment]
exactly, but that is the point. This way they are more likely  to choose the correct term based on the experiment.

**Comment by mah11** [comment]
But there are more experiments that would support "increased distance ...", so "abnormal localization" applies in fewer cases. So why does it make sense to offer the term that fits a narrower selection of experiments first?

**Comment by ValWood** [comment]
They will only see this term if they start from "abnormal process".
I never thought it as being about the number of experiments at all? (I guess I still don't know what you mean here).

I also don't know what you mean about "why does it make sense to offer the term that fits a narrower selection of experiments first".
We aren't offering them this term "first", we have no control over what a user searches on.  If they do end up here and they  really measured "increased distance ..." we want them to be able to see that term, and not to use  "abnormal localization".

if they did not come down through the "abnormal process" branch they would not see the term 
"abnormal localization" so it is not an issue....

**Comment by ValWood** [comment]
Next time you use canto, try to pretend you don't know the ontology and search for a higher level term and drill down ;)
All we are interested in is getting the user to select  *the most appropriate term*. For this, the current arrangement appears to work....

**Comment by mah11** [comment]
> try to pretend you don't know the ontology

sorry, at this point that would require a TBI

**Comment by mah11** [comment]
What I am saying is that the type of experiment done has a kind of bearing on which term is most appropriate. Some experiments simply do not observe processes taking place in real time, and for those experiments the structure phenotype terms are more appropriate.

**Comment by ValWood** [comment]
I think in those cases it is clear that the  output_of terms are not applicable  to use though.
Maybe we do need to think about this some more (perhaps we need to indicate how the term is related), but with current behavior it does appear to work for suggestions ...

We should talk about this at next curator meeting an look at some examples and decide from there.....
